packages/cosmo-dycore/package.py

```diff
@@ -140,4 +140,7 @@ def cmake_args(self):
     @on_package_attributes(run_tests=True)
     def test(self):
       if '+build_tests' in self.spec: 
-            subprocess.run(['./test_dycore.py',  str(self.spec),  self.build_directory], cwd = self.root_cmakelists_dir + '/test/tools')
+          try:
+              subprocess.run(['./test_dycore.py',  str(self.spec),  self.build_directory], cwd = self.root_cmakelists_dir + '/test/tools', stderr=subprocess.STDOUT)
+          except subprocess.CalledProcessError:
+              raise ValueError('Dycore tests failed')
```

Comment:
```suggestion
          subprocess.run(['./test_dycore.py',  str(self.spec),  self.build_directory], cwd = self.root_cmakelists_dir + '/test/tools', check=True, stderr=subprocess.STDOUT)
```
Probably needs `check=True`.

> If check is true, and the process exits with a non-zero exit code, a CalledProcessError exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured.

Source: https://docs.python.org/3/library/subprocess.html?highlight=subprocess#subprocess.run